src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -4360,35 +4785,552 @@ public static void ToLowerInvariant(string s, string expected)
             Assert.Equal(expected, destination.ToString());
         }
 
-        [Theory]
-        [InlineData("")]
-        [InlineData("hello")]
-        public static void ToString(string s)
+        [Fact]
+        public static void ZeroLengthToLower()
         {
-            Assert.Same(s, s.ToString());
-            Assert.Same(s, s.ToString(null));
-            Assert.Same(s, s.ToString(CultureInfo.CurrentCulture));
+            char[] expectedSource = { 'a', 'B', 'c' };
+            char[] a = { 'a', 'B', 'c' };
+            var expectedDestination = new char[1] { 'a' };
+            var expectedDestinationString = new char[0];
+            Span<char> destination = new char[1] { 'a' };
 
-            Assert.Equal(s, s.AsSpan().ToString());
-        }
+            string s1 = new string(a, 2, 0);            
+            Assert.Equal(expectedDestinationString, s1.ToLowerInvariant().ToArray());            
 
-        [Theory]
-        [InlineData("hello", "HELLO")]
-        [InlineData("HELLO", "HELLO")]
-        [InlineData("hElLo", "HELLO")]
-        [InlineData("HeLlO", "HELLO")]
-        [InlineData("", "")]
-        public static void ToUpper(string s, string expected)
-        {
-            Assert.Equal(expected, s.ToUpper());
-            
-            Span<char> destination = new char[s.Length];
-            Assert.Equal(s.Length, s.AsSpan().ToUpper(destination, CultureInfo.CurrentCulture));
-            Assert.Equal(expected, destination.ToString());
+            ReadOnlySpan<char> source = s1.AsSpan();                       
+            Assert.Equal(source.Length, source.ToLower(destination, CultureInfo.CurrentCulture));
+            Assert.Equal(source.Length, source.ToLowerInvariant(destination));
+            Assert.Equal(expectedDestination, destination.ToArray());
+            Assert.Equal(expectedSource, a);
+
+            s1 = string.Empty;            
+            Assert.Equal(expectedDestinationString, s1.ToLowerInvariant().ToArray()); 
+         
+            source = s1.AsSpan();
+            Assert.Equal(source.Length, source.ToLower(destination, CultureInfo.CurrentCulture));
+            Assert.Equal(source.Length, source.ToLowerInvariant(destination));
+            Assert.Equal(expectedDestination, destination.ToArray());
+            Assert.Equal(expectedSource, a);
         }
 
-        private static IEnumerable<object[]> ToUpper_TurkishI_MemberData(
-            params KeyValuePair<char, char>[] mappings)
+        [Fact]
+        public static void SameSpanToLower()
+        {
+            var expected = new char[3] { 'a', 'b', 'c' };
+            var a = new char[3] { 'a', 'B', 'c' };
+            {
+                string s1 = new string(a);                                
```

Comment:
This can be moved outside the inner scopes since it is common.

Reply:
ok